Observer/CancelOrderItemObserver.php

```diff
@@ -150,31 +160,28 @@ public function execute(EventObserver $observer): void
         $this->placeReservationsForSalesEvent->execute($itemsToCancel, $salesChannel, $salesEvent);
 
         $order = $orderItem->getOrder();
-
-        // Source selection happens by default on shipment, so only shipments contain information about the
-        // inventory source.
-        // @TODO add support for multi source stock replenishment on cancellation
-        if ($this->isSingleSourceMode->execute()) {
-            $sourceCode = $this->defaultSourceProvider->getCode();
-
-            $itemsToDeduct = [];
-            foreach ($itemsToCancel as $itemToCancel) {
-                $itemsToDeduct[] = $this->itemToDeductFactory->create([
-                    'sku' => $itemToCancel->getSku(),
-                    'qty' => -$itemToCancel->getQuantity(),
-                ]);
+        $sourceSelectionItems = $this->getSourceSelectionResultFromOrder->execute($order)->getSourceSelectionItems();
```

Comment:
Shouldn't we be loading our source selection items from our custom DB/repository etc? We can load them by order ID?